src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -485,10 +485,39 @@ def update_domains(node: FencilWithTemporaries, offset_provider: Mapping[str, An
                         new_axis = nbt_provider.neighbor_axis.value
                         consumed_domain.ranges.pop(old_axis)
                         assert new_axis not in consumed_domain.ranges
-                        consumed_domain.ranges[new_axis] = SymbolicRange(
-                            im.literal("0", ir.INTEGER_INDEX_BUILTIN),
-                            im.literal(str(horizontal_sizes[new_axis]), ir.INTEGER_INDEX_BUILTIN),
-                        )
+
+                        if symbolic_sizes:
+                            consumed_domain.ranges[new_axis] = SymbolicRange(
+                                im.literal("0", ir.INTEGER_INDEX_BUILTIN),
+                                symbolic_sizes[new_axis],
+                            )
+                        else:
+                            consumed_domain.ranges[new_axis] = SymbolicRange(
+                                im.literal("0", ir.INTEGER_INDEX_BUILTIN),
+                                im.literal(
+                                    str(horizontal_sizes[new_axis]), ir.INTEGER_INDEX_BUILTIN
+                                ),
+                            )
+                    elif offset_provider[offset_name].__class__.__name__ == "DummyConnectivity":
```

Comment:
Please use the `Connectivity` protocol. See comment below in the type inference for more details.